scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,413 @@
+    while (lastIndex >= 0 && artifactName[lastIndex] != ':') {
+      artifactVersion.append(artifactName[lastIndex])
+      lastIndex--
+    }
+    artifactVersion.reverse()
+    val licenseNamesFromPom = mutableListOf<String>()
+    val licenseLinksFromPom = mutableListOf<String>()
+    val licenseList = arrayListOf<License>()
+    val pomFile = URL(pomFileUrl).openStream().bufferedReader().readText()
+    val pomText = pomFile
+    var cursor = -1
+    if (pomText.length > 11) {
+      for (index in 0..(pomText.length - 11)) {
+        if (pomText.substring(index, index + 10) == LICENSES_TAG) {
+          cursor = index + 9
+          break
+        }
+      }
+      if (cursor != -1) {
+        var cursor2 = cursor
+        while (cursor2 < (pomText.length - 12)) {
+          if (pomText.substring(cursor2, cursor2 + 9) == LICENSE_TAG) {
+            cursor2 += 9
+            while (cursor2 < pomText.length - 6 &&
+              pomText.substring(
+                cursor2,
+                cursor2 + 6
+              ) != NAME_TAG
+            ) {
+              ++cursor2
+            }
+            cursor2 += 6
+            val url = StringBuilder()
+            val urlName = StringBuilder()
+            while (pomText[cursor2] != '<') {
+              urlName.append(pomText[cursor2])
+              ++cursor2
+            }
+            while (cursor2 < pomText.length - 4 &&
+              pomText.substring(
+                cursor2,
+                cursor2 + 5
+              ) != URL_TAG
+            ) {
+              ++cursor2
+            }
+            cursor2 += 5
+            while (pomText[cursor2] != '<') {
+              url.append(pomText[cursor2])
+              ++cursor2
+            }
+            val httpUrl = replaceHttpWithHttps(url)
+            licenseNamesFromPom.add(urlName.toString())
+            licenseLinksFromPom.add(httpUrl)
+            licenseList.add(
+              License
+                .newBuilder()
+                .setLicenseName(urlName.toString())
+                .setPrimaryLink(httpUrl)
+                .setPrimaryLinkType(PrimaryLinkType.PRIMARY_LINK_TYPE_UNSPECIFIED)
+                .build()
+            )
+          } else if (pomText.substring(cursor2, cursor2 + 12) == LICENSES_CLOSE_TAG) {
+            break
+          }
+          ++cursor2
+        }
+      }
+    }
+    val mavenDependency = MavenDependency
+      .newBuilder()
+      .setArtifactName(it.coord)
+      .setArtifactVersion(artifactVersion.toString())
+      .addAllLicense(licenseList)
+      .setOriginOfLicense(OriginOfLicenses.UNKNOWN)
+
+    mavenDependencyList.add(mavenDependency.build())
+  }
+  return MavenDependencyList.newBuilder().addAllMavenDependency(mavenDependencyList).build()
+}
+
+private fun replaceHttpWithHttps(
+  urlBuilder: StringBuilder
+): String {
+  var url = urlBuilder.toString()
+  if (url.substring(0, 5) != "https") {
+    url = "https${url.substring(4, url.length)}"
+  }
+  return url
```

Comment:
Suggest using java.net.URL, instead, since you can more easily manipulate parts of it like the scheme or filename (per your above code) before generating a new URL.

Reply:
I tried searching for it on the web, but couldn't find any example. Can you please provide one?

Reply:
Ah, apparently URL and URI are immutable with no builders, so this won't actually work. Please disregard.

However, you can probably simplify this slightly by replace()ing "http" with "https" if the string does not startsWith "https".

Reply:
Done.

Reply:
Verified.
```
    private fun replaceHttpWithHttps(
      urlBuilder: StringBuilder
    ): String {
      var url = urlBuilder.toString()
      if (!url.startsWith("https")) {
        url = url.replace("http", "https")
      }
      return url
    }
```